lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -863,31 +862,91 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
 
         Value axesVal;
         if (!binder.tensorOperandAtIndex(axesVal, 1)) {
-          Torch::BaseTensorType axesType =
-              axesVal.getType().cast<Torch::BaseTensorType>();
-          SmallVector<Value> dimList;
-          SmallVector<int64_t> selectSizes{1};
-          auto selType = rewriter.getType<Torch::ValueTensorType>(
-              selectSizes, axesType.getOptionalDtype());
-          auto axesTy = dyn_cast<Torch::ValueTensorType>(axesVal.getType());
-          auto axesShape = axesTy.getSizes();
+          auto inputType = data.getType().dyn_cast<Torch::ValueTensorType>();
+          if (!inputType.hasSizes() || !resultType.hasSizes()) {
+            return rewriter.notifyMatchFailure(
+                binder.op,
+                "unimplemented: expected input and result to have shapes");
+          }
 
-          if (axesShape.size() != 1 || axesShape[0] == Torch::kUnknownSize)
-            return failure();
+          auto areDistinct = ([](SmallVector<int64_t> array) -> bool {
```

Comment:
I'm not sure if it would be required by some other op lowering. We can keep it here for now and can migrate it to utils if required by some other op lowering in future.